### PR TITLE
[tests-only][full-ci] copy test dependencies from ownCloud/core (stable-2.0)

### PR DESCRIPTION
**Pull request by phil-davis**
## Description
Copy test infrastructure from the oC/core repository

This is a port of #5280 to the `stable-2.0` branch. See discussion in that PR and in the issue for explanation of what is being done.

The API tests no longer use anything from `owncloud/core`

## Related Issue
#5094 

## How Has This Been Tested?
CI

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

>

### CODE REVIEW

**Comment by phil-davis** [comment]
Note: we will leave this here until next week. Also leave #5280 
I will switch it to draft for now.

I would like to double-check that all other test changes have been ported the same into both `master` and `stable-2.0` branches, and get any recent test code changes that have happened in core. And then merge when we are happy that everything is "in sync".

**Comment by saw-jan** [comment]
Todos:
- [ ] update expected failure list

**Comment by saw-jan** [comment]
Changes to notice:
- feature suites are prefixed `core` and copied to the `tests/acceptance/features` folder (e.g. `ApiMain` -> `coreApiMain`)
- separate behat config for copied core tests (tests/acceptance/config/`behat-core.yml`)
- core API tests run in parts

These structures can be discussed.

File: tests/acceptance/run.sh
Comment (phil-davis) [inline tests/acceptance/run.sh:517]: This bit of code might now be useless here in ocis.
But maybe it will be needed (and the if condition modified a bit) if we enable the parallel deployment tests.
So we can leave it here for now.

**Review by phil-davis — COMMENTED** [review]
Looking good to me.

**Review by individual-it — COMMENTED** [review]
I think the structure is fine for now.
From the `.drone.star` file we should be to delete the functions
- cacheCoreReposForTesting
- cloneCoreRepos

**Comment by phil-davis** [comment]
I think that these cleanup things can be done in a next PR after we merge this one. For future comparisons, it is probably cleaner to merge all the feature files as direct copies from core. Then delete unneeded stuff separately. That way the deletions will appear in diff listings that can be easily seen.

- remove any scenarios that are tagged `skipOnOcis`

The core API tests run with these filter tags:
```
~@skipOnGraph&&~@skipOnOcis&&~@notToImplementOnOCIS&&~@toImplementOnOCIS&&~comments-app-required&&~@federation-app-required&&~@notifications-app-required&&~systemtags-app-required&&~@local_storage&&~@skipOnOcis-OCIS-Storage&&~@caldav&&~@carddav&&@api&&~@skip
```
- look through the filter tags and sort out which ones we still care about. 
- remove scenarios for any tags that we know are never gong to be relevant to ocis or reva
- cut-down the filter-tags list

**Comment by phil-davis** [comment]
The core-api CI pipelines look good - they run suites and scenarios without ever fetching anything from the `owncloud/core` repo.

**Comment by saw-jan** [comment]
This PR is ready from my side.

**Comment by phil-davis** [comment]
Note: these test suites have not been copied from core:
apiComments - comments will not work like that in ocis anyway, if and when implemented
apiFederation* - "core federation" is an oC10 thing
apiProvisioning-v* - we don't use the provisioning API any more
apiProvisioningGroups-v* - we don't use the provisioning API any more
apiShare*ToRoot* - these are for shares received directly in the sharees root folder. That does not happen in ocis
apiSharingNotifications* - they are specific to the notifications app behavior in oC10
apiTags - that is a specific implementation of tags for oC10

That looks good to me.

**Comment by phil-davis** [comment]
> I think the structure is fine for now. From the `.drone.star` file we should be to delete the functions
> 
>     * cacheCoreReposForTesting
> 
>     * cloneCoreRepos

Those are now only used by `parallelDeployAcceptancePipeline` - we could remove them now, or we leave them as-is and sort out the needed refactoring when the `parallelApiTests` are enabled again "some day".

**Comment by individual-it** [comment]
as discussed in the standup meeting, parallel deployment tests are not prio now, so lets delete `cacheCoreReposForTesting` & `cloneCoreRepo` now and the parallel deployment tests in an other PR as its dead code. We can bring it back when its needed.

**Comment by phil-davis** [comment]
> as discussed in the standup meeting, parallel deployment tests are not prio now, so lets delete `cacheCoreReposForTesting` & `cloneCoreRepo` now and the parallel deployment tests in an other PR as its dead code. We can bring it back when its needed.

I will push another commit to cleanup.

**Comment by phil-davis** [comment]
@individual-it @micbar this is ready for review. We need 2 approvals to merge. I have reviewed the code that was done by @kiranparajuli589 and @saw-jan but I cannot approve because I created the PR.

There are many thousands of lines because lots of code and feature files have been copied from `owncloud/core` - I don't expect all the lines to be individually reviewed!

**Review by individual-it — APPROVED** [review]

**Comment by ScharfViktor** [comment]
looks cool 
questions: should we clean up some of the moved tests? for example: tagsTests or tets with `@skipOnOcis` tag 
Or will we clean it up later in another PR?

**Comment by phil-davis** [comment]
> Or will we clean it up later in another PR?

I suggest that we clean up in another PR - see my comment https://github.com/owncloud/ocis/pull/5306#issuecomment-1370606999

**Comment by ScharfViktor** [comment]
yes, sorry. Wasn't paying attention.
then in order not to lose what i found. need remove:
- TagsContext
- TagsHelper
- TranslationHelper
- translation.feature

**Review by ScharfViktor — APPROVED** [review]

**Pull request merged**
